rpcs3/Input/evdev_gun_handler.cpp

```diff
@@ -0,0 +1,273 @@
+// This makes debugging on windows less painful
+//#define HAVE_LIBEVDEV
+
+#ifdef HAVE_LIBEVDEV
+
+#include "evdev_gun_handler.h"
+#include <libudev.h>
+#include <fcntl.h>
+#include <linux/input.h>
+
+LOG_CHANNEL(evdev_log, "evdev");
+
+evdev_gun_handler* evdev_gun_handler_instance = NULL;
```

Comment:
This whole global singleton logic is very obsolete.
Just use g_fxo for this purpose.

Reply:
global variables like g_fxo is even more obsolete in fact. it is a very subjective.

Reply:
What I mean is: we use g_fxo instead of random globals, because this whole singleton/shutdown logic you added is handled by the construction and destruction of the fxo automatically